pkg/nvml/gpm.go

```diff
@@ -44,23 +44,23 @@ func (l *library) GpmSampleAlloc(GpmSample *GpmSample) Return {
 
 // nvml.GpmSampleGet()
 func (l *library) GpmSampleGet(Device Device, GpmSample GpmSample) Return {
-	return nvmlGpmSampleGet(Device, GpmSample)
+	return Device.GpmSampleGet(GpmSample)
 }
 
-func (Device Device) GpmSampleGet(GpmSample GpmSample) Return {
-	return GpmSampleGet(Device, GpmSample)
+func (Device nvmlDevice) GpmSampleGet(GpmSample GpmSample) Return {
+	return nvmlGpmSampleGet(Device, GpmSample)
 }
 
 // nvml.GpmQueryDeviceSupport()
 type GpmSupportV struct {
-	device Device
+	device nvmlDevice
 }
 
 func (l *library) GpmQueryDeviceSupportV(Device Device) GpmSupportV {
```

Comment:
This is probably out of scope, but does the `GpmQueryDeviceSupportV` top-level function (and others like it) add any value? Should we not *always* use the `GpmQueryDeviceSupportV` member function?

Reply:
No, you don't need to. If you just want the latest, you call e.g. `GpmQueryDeviceSupport()` directly. Only if you care about a specific version do you call `GpmQueryDeviceSupportV().V1()`, `GpmQueryDeviceSupportV().V2()`, etc.